ls1intum/Artemis
# [Enhancement] Show test names to students

### Checklist

### Motivation and Context



Resolves #2359 

### Description


- Add a new 'showTestNamesToStudents' property to the programming exercise entity
- Use this property in all places where the result-detail component is opened
  - When enabled, the student effectively sees the same information on feedbacks as the instructor
  - (The instructor always sees the test names, regardless of the option)
- Some refactoring for the modal creation to have proper type support.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a Programming Exercise with the new option enabled
4. Participate as a student and view the feedback, test case names should be visible and test case feedback should be distinguishable from manual feedback.
5. Edit the exercise to disable the new option
6. View again as a student, there should not be any visual distinction between manual and automatic feedback or any test case names shown
7. View as an instructor, test case names should be visible and test case feedback should be distinguishable from manual feedback.
8. Delete the exercise.

### Test Coverage




**Client:**



### Screenshots



**Exercise Creation Option**
<img width="441" alt="Bildschirmfoto 2020-11-12 um 16 40 57" src="https://user-images.githubusercontent.com/13920539/98962586-1313de00-2507-11eb-87b2-0bfd0bf9f013.png">

**With option enabled** (Same as instructor view)
<img width="787" alt="Bildschirmfoto 2020-11-12 um 16 48 52" src="https://user-images.githubusercontent.com/13920539/98962628-1f983680-2507-11eb-892f-726e21285b90.png">

**With option disabled** (Default)
<img width="787" alt="Bildschirmfoto 2020-11-12 um 16 49 17" src="https://user-images.githubusercontent.com/13920539/98962703-2f177f80-2507-11eb-8115-e1b4a9e587f5.png">
